mlflow/pyfunc/__init__.py

```diff
@@ -624,13 +624,15 @@ class PyFuncModel:
     ``model_meta`` contains model metadata loaded from the MLmodel file.
     """
 
-    def __init__(self, model_meta: Model, model_impl: Any):
+    def __init__(self, model_meta: Model, model_impl: Any, pyfunc_predict_func: str = "predict"):
         if not hasattr(model_impl, "predict"):
             raise MlflowException("Model implementation is missing required predict method.")
         if not model_meta:
             raise MlflowException("Model is missing metadata.")
         self._model_meta = model_meta
         self._model_impl = model_impl
+        assert hasattr(model_impl, pyfunc_predict_func)
```

Comment:
Can we `raise MlflowException` if the predict function isn't available instead of `assert`?